packages/data-point/docs/best-practices.md

```diff
@@ -0,0 +1,457 @@
+
+**Example:**
+
+```js
+const DataPoint = require('data-point')
+const entities = {
+  'model:Card': {
+    value: {
+      id: '$uid',
+      title: '$shortTitle',
+      date: localizeDate('lastModified')
+    }
+  }
+}
+const dp = DataPoint.create({
+  entities, // registering entities
+})
+
+// call entity by registered reference
+dp.resolve('model:Card', input)
+  .then(result => {
+    console.dir(result)
+  })
+```
+
+#### Instance Entity
+
+It is designed to be used as a stand-alone reducer that can be declared on a module and exported to be used elsewhere. Because it is the result of an entity factory it produces a variable which can easily be looked up using lookup functionality on most modern IDEs. (e.g. [vscode go-to-definition](https://code.visualstudio.com/docs/editor/editingevolved#_go-to-definition)).
+
+**Example:**
+
+```js
+const DataPoint = require('data-point')
+
+// card is a variable you can export from a module
+const Card = DataPoint.Model('Card', {
+  value: {
+    id: '$uid',
+    title: '$shortTitle',
+    date: localizeDate('lastModified')
+  }
+})
+
+const dp = DataPoint.create()
+
+// we call the entity directly
+dp.resolve(Card, input)
+  .then(result => {
+    console.dir(result)
+  })
+```
+
+### Performance
+
+Both entities internally are exactly the same, the main difference is how DataPoint access them when they are executed. Registered entities need to be looked up, which usually takes nanoseconds depending on how many entities are registered and could be considered the only real disadvantage compared to an entity instance.
+
+#### Feature comparison
+
+| Usage | Registered | Instance |
+|:--|:--:|:--:|
+| Testable | ✔️ | ✔️ |
+| Globally available | ✔️ | ✖️ |
+| Suited to be exported as part of a module | ✖️ | ✔️ |
+| Discoverability | ✖️ | ✔️ |
+| Compilation check | ✖️ | ✔️ |
+| Runtime check | ✔️ | ✔️ |
+
+## Testing
+
+Testing DataPoint transformations should not be any different from any other 
+
+### Reducer Paths
+
+[Reducer paths](https://github.com/ViacomInc/data-point/tree/master/packages/data-point#path-reducer) are the most basic reducer and probably they not worth testing individually, if what you want is to test data integrity you should probably either use the Entity's [type checking](https://github.com/ViacomInc/data-point/tree/master/packages/data-point#entity-type-check) hooks or/and use the [Schema Entity](https://github.com/ViacomInc/data-point/tree/master/packages/data-point#schema-entity) for a full JSONSchema validation.
+
+### Function Reducer
+
+[Function Reducer](https://github.com/ViacomInc/data-point/tree/master/packages/data-point#function-reducer) should be tested as any other Javascript function, the singularity of a reducer is its second parameter the accumulator which at times you might need to mock the Accumulator object to be able to test your reducer. On most cases testing a function reducer does not need an instance of DataPoint to test its functionality.
```

Comment:
>the singularity of a reducer

I'm not sure what singularity means here, please use smaller words so . Ican understand better; also, this could link down to the section that actually describes how to mock the accumulator

Reply:
done! thanks!